File: src/main/java/me/miki/shindo/management/mods/settings/impl/ImageSetting.java
package me.miki.shindo.management.mods.settings.impl;

import me.miki.shindo.Shindo;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.mods.Mod;
import me.miki.shindo.management.mods.settings.Setting;

import java.io.File;

public class ImageSetting extends Setting {

    private File image;

    public ImageSetting(TranslateText nameTranslate, Mod parent) {
        super(nameTranslate, parent);

        this.image = null;

        Shindo.getInstance().getModManager().addSettings(this);
    }

    @Override
    public void reset() {
        this.image = null;
    }

    public File getImage() {
        return image;
    }

    public void setImage(File image) {
        this.image = image;
    }
}

File: src/main/kotlin/com/shindoclient/shindo/management/settings/impl/ImageSetting.kt
package com.shindoclient.shindo.management.settings.impl

import com.shindoclient.shindo.management.language.TranslateText
import com.shindoclient.shindo.management.settings.Setting
import com.shindoclient.shindo.management.settings.config.ConfigOwner
import java.io.File

open class ImageSetting : Setting {
    private var image: File? = null

    constructor(nameTranslate: TranslateText, parent: ConfigOwner) : super(nameTranslate, parent)

    constructor(name: String, parent: ConfigOwner) : super(name, parent)

    override fun reset() {
        image = null
    }

    fun getImage(): File? = image

    open fun setImage(image: File?) {
        this.image = image
    }
}
